evap/staff/views.py

```diff
@@ -1173,7 +1173,7 @@ def notify_reward_points(users, **kwargs):
     user = get_object_or_404(UserProfile, id=user_id)
     form = UserForm(request.POST or None, request.FILES or None, instance=user)
 
-    courses_contributing_to = Course.objects.filter(semester=Semester.active_semester(), contributions__contributor=user)
+    courses_contributing_to = [(semester, Course.objects.filter(semester=semester, contributions__contributor=user)) for semester in Semester.objects.all()]
```

Comment:
this includes all the semesters, even if the contributor had to courses in it. the result could be rather ugly in the UI

Reply:
it's actually filtered in the ui. however it would be better to not add them at all.

Reply:
to me it looks like the filtering needs to be done only once, so it can actually be moved from the template to here. and since it's not more complexity doing it here than there, i don't see why we should carry data around that's never used anyways and not filter it right at the place where it's produced.